Sources/Client/Model/Channel/Channel.swift

```diff
@@ -243,7 +243,7 @@ extension Channel {
             extraData?.name
         }
         set {
-            var object: ChannelExtraDataCodable = extraData ?? ChannelExtraData()
+            var object: ChannelExtraDataCodable = extraData ?? ChannelExtraData(name: newValue, imageURL: nil)
             object.name = newValue
             extraData = object
```

Comment:
Don't be lazy :)
```suggestion
            if let object = extraData {
                var object = object
                object.name = newValue
                extraData = object
            } else {
                extraData = ChannelExtraData(name: newValue, imageURL: nil)
            }
```

We need PropertyWrapper: `@ExtraDataProperty`!
```
    @ExtraDataProperty
    var name: String?
```
or! It will be better for developers
```
    @ExtraDataProperty(type: ChannelExtraData.self)
    var name: String?
```

Reply:
I'd prefer
```swift
var object: ChannelExtraDataCodable = extraData ?? ChannelExtraData(name: nil, imageURL: nil)
```
then
```swift
 if let object = extraData {
                var object = object
```

for propertyWrapper, I'm not sure what its function will be.